### ISSUE TITLE
Hardware revision request: Improved power efficiency

### ISSUE THREAD

**Issue by adamgarbo** [sparkfun/MicroMod_Data_Logging_Carrier#3]
Hi there,

I've been super excited to get my hands on the MicroMod Data Logging Carrier Board (MMDLC). I've been putting it through its paces along with an Artemis Processor. 

Unfortunately, the quiescent draw of the MMDLC is quite high, so my goal has been to track down the major current draw contributors in order to provide constructive feedback for the next hardware revision (which I realize may take some time). I believe with a few minor changes, the power efficiency of this carrier board could be really improved!

### Methods: Current Measurement
The first step was to characterize the current draw of the MMDLC under a number of different conditions (e.g. disabling LEDs, voltage divider, etc). My methods for measuring the current were as follows:

1. Charge RTC battery for >24 hours via USB-C/LiPo battery
2. Connect a 3.7 V lithium battery (4.2 V nominal), disconnect USB-C connection and allow LiPo charger circuit to settle
3. Place system in deep sleep using the Apollo3 Core RTC library example: `Example6 _LowPower_Alarm`  with added functionality to correctly power down I2C/SPI and G1/G2 pins 
4. Disable the onboard voltage divider using a hot-air rework station to remove the top 20kΩ resistor 
5. All current measurements were made across MEAS jumper with a Fluke 287 digital multimeter

### Results:

**Table 1.** Quiescent draw of MicroMod Data Logger Carrier Board & Artemis Processor using a 3.7 V (4.2 V) LiPo battery
3V3 LED | VIN LED | PWC | Qwiic | Voltage Divider | Quiescent Draw (mA)
:--: | :--: | :--: | :--: | :--: | :--:
TRUE | TRUE | FALSE | FALSE | TRUE | 2.570
FALSE | TRUE | FALSE | FALSE | TRUE | 0.930
FALSE | FALSE | FALSE | FALSE | TRUE | 0.475
FALSE | FALSE | FALSE | FALSE | FALSE | 0.290

### Discussion:

- Based on the results of Table 1, disabling both LEDs brings the quiescent draw to approximately 500 uA, which is a good start. 
- The 20kΩ/10kΩ voltage divider was identified in issue #1 as drawing as much as 200 uA (constant) at 6 V. In this case, disabling the voltage divider by removing the top SMD resistor appeared to save 185 uA. A good savings, but not a modification the average user can will/can do.
- The lowest quiescent draw measured was approximately 290 uA. This is approximately 3-4x greater than the sleep current of other comparable data logging platforms, such as the Adafruit M0 Adalogger (~80 uA) or the SparkFun OpenLog Artemis (~80 uA).  
- The question is, what are the sources of this 290 uA? Looking at the schematic and related datasheets, I was able to come up with the following values:

**Table 2.** Max quiescent draw of MMDLCB components
Component | Max Current Draw (uA)
:--: | :--:
AP7361C LDO Regulator | 80
PWC AP2112 LDO Regulator | 80
Qwiic AP2112 LDO Regulator | 80
MCP7381 LiPo Charger | 5?

Based on my experience, the AP2112 LDO regulators will draw around 60 uA. Adding this value for the PWC and Qwiic LDOs with the new AP7361C LDO and MCP7381 charger, I account for approximately 180 uA. This leaves another 100 uA being drawn from components I haven't yet been able to identify. 

### Suggestions:
While I'm still trying to identify the other contributors to increased sleep current draw, my hope is to get the conversation started regarding potential modifications that could help to improve the overall power efficiency. A couple of suggestions include:

1. Add (back) P-channel MOSFETs to provide control over PWC/Qwiic regulators. Potential power savings: 160 uA.
    - I noticed in a previous version of the MMDLCB that there were MOSFETs present and in fact, the Hookup Guide still makes reference to them.
2. Voltage divider:
    - (Option 1) Add jumper pad to voltage divider to allow it to be disabled. Potential power savings: 200 uA.
    - (Option 2) Add MOSFET to voltage divider allowing it to be enabled/disabled using a GPIO pin. @PaulZC's Global Artemis Tracker does something similar with the very same voltage divider resistance values:

![image](https://user-images.githubusercontent.com/22924092/98469824-ddda5980-21af-11eb-9382-265d452f2ff3.png)

### Summary:
The MicroMod Data Logging Carrier Board has amazing potential and it'd be super if we could get this baby under 100 uA! 

Cheers,
Adam

**Comment by adamgarbo** [comment]
It just dawned on me that the two AP2112 LDO regulators would only draw a maximum of 1 uA when EN is pulled LOW, which means that there's no real need for P-MOSFETs to disable the power. :man_facepalming:

![image](https://user-images.githubusercontent.com/22924092/99576963-aab27a00-29a8-11eb-9771-c4f51b656e18.png)

However, this has left me even more confused as to where all the quiescent draw is coming from. Perhaps I should be looking at the Artemis Processor? The investigation continues!

**Comment by jurvanoerle** [comment]
Hi there Adam,
I have been trying to do the same power-measurements ( (MMDLC with an Artemis Processor), but I find much higher values.
With all leds disabled, and G0,G1 and G2 disabled (voltage divider in place) I find values >> 1 mA. (and fluctuating). Obviously, I'm doing something wrong.    I have 2 questions:
= could you show your arduino code so I can retry your measurements, and
= where can I find the resistor of the voltage divider I need to cut to reduce power

Regards, Jur

**Comment by adamgarbo** [comment]
Hi @jurvanoerle,

Happy to share my code. It's essentially an adapted version of the Apollo3 Example6 _LowPower_Alarm. You can access it here: 
https://github.com/adamgarbo/Cryologger_Iceberg_Tracking_Beacon/blob/release-candidate/Software/mmdlcb_artemis_low_power/mmdlcb_artemis_low_power.ino

Please keep in mind that the RTC battery backup onboard the MMDLC takes a *long* time to fully charge. I've found it takes more than 24 hours for the battery to completely charge and for the quiescent draw to achieve its lowest value. Your current draw will be much higher (and fluctuate) while it is charging. Also note, as soon as you remove all power sources from the MMDLC, the battery appears to rapidly discharge itself and will require charging again.

I used a hot-air rework station to remove the top SMD resistor from the voltage divider (green arrow). If you prefer to cut the trace (**Warning: permanent change**), you can cut the small trace below the resistor marked by the yellow arrow. Note: be careful not to cut the wider trace below the resistor, as this is your VBAT/VIN rail.

![image](https://user-images.githubusercontent.com/22924092/100107822-13ca3f80-2e38-11eb-97ab-657cebcd1727.png)

Cheers,
Adam

**Comment by adamgarbo** [comment]
Oh and before I forget, I believe I may have discovered the source of the high quiescent draw I observed. The CMOS op-amp on the Artemis Processor:

![image](https://user-images.githubusercontent.com/22924092/100108870-46c10300-2e39-11eb-81a6-e0592c0fe096.png)

As best I can tell, this baby draws 31-50 uA per channel plus 45 uA quiescent draw to power itself. With four of the op-amp channels in use, this could explain the power draw measured with the MMDLCB and Artemis Processor. Hopefully, @nseidle can let us know if this is correct!

Cheers,
Adam

**Comment by jurvanoerle** [comment]
Hi Adam,
I managed to brick my board removing the divider/resistor. A new board shows much better results.
The rtc battery discharges quickly indeed after removing all power: the processor and even I2C and leds are powered by the rtc battery, discharging in seconds. For a datalogger board you would expect that the rtc-battery only powers the rtc, lasting for month's: how can I prevent this discharge (cutting a trace somehwere?)
Regards, Jur

**Comment by adamgarbo** [comment]
Hi Jur,

Sorry to hear about your board becoming bricked. I'm trying to think about how that could have happened! The voltage divider is in a pretty non-critical area. I've posted an image of the EAGLE board file with what should have been cut or removed in purple. I'm partial to removing the SMD resistor, as it can always be soldered back on.

Regarding the RTC battery backup discharging quickly, I think this will be something that also needs to be addressed in the next hardware revision. I get the sense that the MicroMod ecosystem is still in its infancy, and still has a lot of kinks to be worked out.

Cheers,
Adam 

![image](https://user-images.githubusercontent.com/22924092/100393361-5d1cb980-3007-11eb-82e3-ab37cf77bfc5.png)

**Comment by jurvanoerle** [comment]
Hi Adam,
I was succesfull in getting comparable results ( ~0,3 mA in deepsleep), which is perfect for my application, thanks. 
BTW, I managed to brick my second carrier board also. On both boards, the serial port is not showing up anymore (not on a PC, nor on a Linux laptop). Is it possible to brick the fdti chip by software (eg by shutting down serial, or shutting down G0/G1/G2 in the arduino code; I have been experimenting a lot with these settings). 
I have never experienced this with other arduino boards. 
Regards, Jur

**Comment by adamgarbo** [comment]
To provide a quick update, I measured the current draw of the MMDLCB with no processor board installed and powered by a 4.2V LiPo.

I made two measurements, as the AP2112 regulators are enabled by default, which required manually pulling their EN pins LOW.

3V3 LED | VIN LED | PWC | Qwiic | Voltage Divider | Quiescent Draw (uA)
-- | -- | -- | -- | -- | --
FALSE | FALSE | TRUE | TRUE | FALSE | 200
FALSE | FALSE | FALSE | FALSE | FALSE | 149

These current draws look to be accounted for by the AP7361C regulator (~80 uA) and the two 100 kΩ pull-up resistors on the AP2112 regulator EN pins (66 uA).

Doing the math, it looks like the Artemis Processor is consuming ~140 uA on its own.

While it'd be nice to see the AP2112s behind FETs, adding a jumper to the voltage divider is by far the easiest solution and would be ideal for all MicroMod carrier boards.
